newIDE/app/src/MainFrame/Preferences/PreferencesProvider.js

```diff
@@ -268,17 +270,45 @@ export default class PreferencesProvider extends React.Component<Props, State> {
     }
   }
 
-  _persistValuesToLocalStorage(preferences: Preferences) {
+  _savePreferencesValues(values: PreferencesValues) {
     try {
-      localStorage.setItem(
-        LocalStorageItem,
-        JSON.stringify(preferences.values)
-      );
+      localStorage.setItem(LocalStorageItem, JSON.stringify(values));
     } catch (e) {
       console.warn('Unable to persist preferences', e);
     }
+  }
+
+  _loadLatestPath(project: gdProject, kind: ResourceKind) {
+    try {
+      const projectName = project.getName();
+      const values = this._loadPreferencesValues();
+      if (values) {
+        const curProjectPaths = values.latestPath[projectName];
+        if (curProjectPaths) {
+          console.log(curProjectPaths, curProjectPaths[kind]);
+          return curProjectPaths[kind];
+        }
+      }
+    } catch (e) {
+      console.warn('Unable to load latest path', e);
+    }
+  }
 
-    return preferences;
+  _saveLatestPath(project: gdProject, kind: ResourceKind, path: string) {
```

Comment:
Passing `project: gdProject` and `kind: ResourceKind` seems a great idea 👍
This allow to encapsulate in this function how we save things internally.